### PR TITLE
feat(core): Add abortOnError option in NestApplicationContextOptions

### PR DESCRIPTION
**Pull request by princechauhan1992**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

https://github.com/nestjs/nest/issues/5162

Issue Number: 5162

## What is the new behavior?
An abortOnError flag is added in nestApplicationContext. By default the process is aborted, but if abortOnError is set to false,
error is rethrown instead of calling process.abort() or process.exit(1)

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by princechauhan1992** [comment]
@kamilmysliwiec  This is my first contribution to an open source project, so I apologize in advance in case I have made any mistakes. Kindly suggest if any changes are required.

File: packages/common/interfaces/nest-application-context-options.interface.ts [hunk 1]
```diff
@@ -8,4 +8,11 @@ export class NestApplicationContextOptions {
    * Specifies the logger to use.  Pass `false` to turn off logging.
    */
   logger?: LoggerService | LogLevel[] | boolean;
+
+  /**
+   * Whether to abort the process on Error. By default the process is exited.
+   * Pass `false` to override default behaviour. If `false` is passed nest will not exit
+   * the application on an exception and instead, will throw the exception.
+   */
```
Comment (kamilmysliwiec) [inline packages/common/interfaces/nest-application-context-options.interface.ts:16]:
```suggestion
  /**
   * Whether to abort the process on Error. By default, the process is exited.
   * Pass `false` to override the default behavior. If `false` is passed, Nest will not exit
   * the application and instead, will rethrow the exception.
   */
```

File: packages/common/interfaces/nest-application-context-options.interface.ts [hunk 2]
```diff
@@ -8,4 +8,11 @@ export class NestApplicationContextOptions {
    * Specifies the logger to use.  Pass `false` to turn off logging.
    */
   logger?: LoggerService | LogLevel[] | boolean;
+
+  /**
+   * Whether to abort the process on Error. By default, the process is exited.
+   * Pass `false` to override the default behavior. If `false` is passed, Nest will not exit
+   * the application and instead will rethrow the exception.
+   */
+  abortOnError?: boolean | null;
```
Comment (kamilmysliwiec) [inline packages/common/interfaces/nest-application-context-options.interface.ts:17]:
```suggestion
  abortOnError?: boolean | undefined;
```

**Comment by nikxco** [comment]
I'm using NestJS 10.0.0 and still having the same issue.
Following is my app module where i'm using yup to validate my environment variables which is working fine except the ValidationError doesn't have all the information.
`@Module({
  imports: [
    ConfigModule.forRoot({
      isGlobal: true,
      cache: false
      expandVariables: true,
      envFilePath: ['./env/.env.local', './env/.env.database'],
      load: [databaseConfig, appConfig],
      validationSchema: EnvironmentVariableSchema,
      validationOptions: {
        abortEarly: false,
      },
    }),
    MongooseModule.forRootAsync({
      useFactory: async (configService: ConfigService) => ({
        dbName: configService.get<string>('database.name'),
        uri: configService.get<string>('database.uri'),
      }),
      inject: [ConfigService],
    }),
    UsersModule,
    DatabaseModule,
    VerificationsModule,
    SubscriptionsModule,
    JobsModule,
    ResumesModule,
    LocationsModule,
    LanguagesModule,
    AlertsModule,
    CampaignsModule,
    CategoriesModule,
    CompaniesModule,
    CoursesModule,
    CreditsModule,
    CurrenciesModule,
    DesignationsModule,
    InsightsModule,
    InstitutionsModule,
    NextAuthModule,
    NotificationsModule,
    OpenaiModule,
    ReportsModule,
    RewardsModule,
    TagsModule,
  ],
  controllers: [AppController],
  providers: [
    AppService,
    { provide: 'APP_NAME', useValue: 'NestJS' },
    { provide: 'APP_VERSION', useValue: '1.0.0' },
    { provide: 'APP_GUARD', useClass: ApiKeyGuard },
  ],
})
export class AppModule {}`

main.ts file
`async function bootstrap() {
  const logger = new Logger('Bootstrap');
  const app = await NestFactory.create(AppModule, { abortOnError: false });
  const configService = app.get(ConfigService);
  const PORT = configService.get('app.apiPort', 3000);
  app.setGlobalPrefix('v1');
  app.useGlobalInterceptors(new LoggingInterceptor());
  app.useGlobalFilters(new MongoExceptionFilter());
  await app.listen(PORT);
  logger.log(`Application is running on: http://localhost:${PORT}`);
}
bootstrap().catch((error) => {
    //This catch block never gets executed.
    console.error('Error in bootstrap++++++++++++++++++++++++', error);
  });`
  
  I want to log bootstrap errors to an external system and in order to do so I must be able to to catch those errors explicitly.
  
  Am I missing something here?? Please help me with this. @kamilmysliwiec

**Pull request merged**